pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -345,16 +347,14 @@ func (l *LibvirtDomainManager) setGuestTime(vmi *v1.VirtualMachineInstance) erro
 	// environment, especially QEMU agent presence) or that the set time is
 	// very precise (NTP in the guest should take care of it if needed).
 
-	domName := api.VMINamespaceKeyFunc(vmi)
-	dom, err := l.virConn.LookupDomainByName(domName)
-	if err != nil {
-		log.Log.Object(vmi).Reason(err).Error(failedSyncGuestTime)
-		return err
-	}
-
-	go func() {
+	go l.setTimeOnce.Do(func() {
```

Comment:
I think the recommended way of doing something like that is:
```go
	l.setTimeOnce.Do(func() {
		go func() {
			// Actual logic
		}()
	})
```

The difference is that with the current code IIUC a new go routine will always be triggered, but will serve as a no-op after the first execution, while with my suggestion only a single go routine would ever be triggered.

Reply:
Nice, thank you for the suggestion, def better!